core/corehttp/gateway_handler.go

```diff
@@ -162,7 +116,7 @@ func (i *gatewayHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		errmsg = errmsg + "bad request for " + r.URL.Path
 	}
 	w.Write([]byte(errmsg))
-	log.Debug(errmsg)
+	log.Error(errmsg) // TODO(cryptix): Why are we ignoring handler errors?
```

Comment:
> any time an error is ignored anywhere in our code, its a bug.

very much agreed.

though:

1. logging is not always an ignored error. sometimes there is nothing to do in a messy io environment (e.g. receiving incoming connections from things that are not running the ipfs protocol. there is nothing to do but log it).
2. this error isn't ignored. an HTTP error header is written, and the error itself is written out to the user for feedback. we could do something else for bookkeeping (like add it to a counter somewhere, throw an event, etc), but correctness is ok here.